socorro/external/boto/connection_context.py

```diff
@@ -14,20 +14,78 @@
 from configman import Namespace, RequiredConfig, class_converter
 
 from socorrolib.lib.converters import change_default
+from socorrolib.lib.ooid import dateFromOoid
 
 
 class KeyNotFound(Exception):
     pass
 
 
 class JSONISOEncoder(json.JSONEncoder):
-
     def default(self, obj):
         if isinstance(obj, datetime.date):
             return obj.isoformat()
         raise NotImplementedError("Don't know about {0!r}".format(obj))
 
 
+class KeyBuilderBase(object):
+    """Base key builder for s3 pseudo-filenames"""
+    def build_keys(self, prefix, name_of_thing, id):
+        """
+        Use S3 pseudo-directories to make it easier to list/expire later
+        {prefix}/v1/{name_of_thing}/{id}
+        """
+        return [
+            '%s/v1/%s/%s' % (prefix, name_of_thing, id)
+        ]
+
+
+class DatePrefixKeyBuilder(KeyBuilderBase):
+    """s3 pseudo-filename key builder with date prefixes
+
+    This adds the date to the prefix for raw_crash objects making it easier to
+    list all the raw_crash objects for a specific date.
+
+    """
+    def build_keys(self, prefix, name_of_thing, id):
+        """Use S3 pseudo-directories to make it easier to list/expire.
+
+        For "raw_crash" things, this pulls out a date and adds that as an
+        additional part of the directory structure::
+
+            {prefix}/v2/{name_of_thing}/{id-3chars}/{date}/{id}
+
+        Then it also returns the old-style key so that we can fetch objects
+        that were saved with the old key::
+
+            {prefix}/v1/{name_of_thing}/{id}
+
+        """
+        keys = []
+
+        if name_of_thing == 'raw_crash':
+            datestamp = dateFromOoid(id)
+
+            if datestamp is not None:
+                # We insert the first 3 chars of the ooid which are hex
```

Comment:
Can't answer with confidence. But what I can say with confidence is that we uses to refer to it as "ooid" and later changed it to "uuid" and later changed it again to "crash ID". There was a point where we pondered to rename the postgres column to be called "crash_id" (instead of "uuid") but an ALTER TABLE command like that would be deemed possibly too time expensive on such a crucial table. New postgres things calls it "crash_id" instead. 

Just to make things even more confusing, see https://github.com/mozilla/socorro/blob/master/socorro/external/postgresql/models.py#L1056 :)

But basically, if it's a comment of English it should ideally be "crash ID".